packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -993,7 +993,6 @@ function completeUnitOfWork(workInProgress: Fiber): Fiber | null {
 
       if (nextUnitOfWork !== null) {
         // Completing this fiber spawned new work. Work on that next.
-        nextUnitOfWork.firstEffect = nextUnitOfWork.lastEffect = null;
```

Comment:
So this was the thing that caused us to "forget" about deletions?

Reply:
Yes